modal-go/cls.go

```diff
@@ -0,0 +1,215 @@
+package modal
+
+// Cls lookups and Function binding.
+
+import (
+	"context"
+	"fmt"
+
+	pb "github.com/modal-labs/libmodal/modal-go/proto/modal_proto"
+	"google.golang.org/protobuf/proto"
+)
+
+type Cls struct {
+	ClassId           string
+	methods           map[string]*Function
+	schema            []*pb.ClassParameterSpec // schema for parameters used in init
+	serviceFunctionId string
+	ctx               context.Context
+}
+
+// Creates parametrized instnace of Cls.
+func (c *Cls) Instance(kwargs map[string]any) (*Cls, error) {
+	if len(c.schema) == 0 {
+		return c, nil
+	} else {
+
+		// Create a parameter set with values from kwargs
+		paramSet := pb.ClassParameterSet_builder{
+			Parameters: []*pb.ClassParameterValue{},
+		}.Build()
+
+		// For each parameter in the schema
+		for _, paramSpec := range c.schema {
+			name := paramSpec.GetName()
+			value, provided := kwargs[name]
+
+			// Skip if not provided and has default value
```

Comment:
We shouldn't skip parameters with defaults - instead it insert the default value into the parameter set - otherwise a defaulted parameter and a literal value matching the default will end up spawning different containers

Reply:
Sounds good. Done.